### PR TITLE
hsrp_global and interface hsrp

### PR DESCRIPTION
**Pull request by saichint**
This PR is for adding cisco_hsrp_global provider and also adding hsrp properties to cisco_interface provider. 
All tests pass on all relevant platforms.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1905,6 +1910,35 @@ ID of the VLAN(s) tha are members of this topology. Valid values are integer/int
...
+Manages a Cisco Hot Standby Router Protocol (HSRP) global parameters.
+
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I2(1)        | 1.5.0                  |
+| N3k      | 7.0(3)I2(1)        | 1.5.0                  |
+| N5k      | 7.3(0)N1(1)        | 1.5.0                  |
+| N6k      | 7.3(0)N1(1)        | 1.5.0                  |
+| N7k      | 7.3(0)D1(1)        | 1.5.0                  |
+| N8k      | 7.0(3)F1(1)        | 1.5.0                  |
```
Comment (mikewiebe) [inline README.md]: Update `N8k` references to be `N9k-F`

File: README.md [hunk 2]
```diff
@@ -1905,6 +1910,35 @@ ID of the VLAN(s) tha are members of this topology. Valid values are integer/int
 Descriptive name of the topology. Valid values are string
 
 --
+### Type: cisco_hsrp_global
+
+Manages a Cisco Hot Standby Router Protocol (HSRP) global parameters.
+
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I2(1)        | 1.5.0                  |
```
Comment (mikewiebe) [inline README.md]: Did we verify this all the way back to `I2` images?

File: README.md [hunk 3]
```diff
@@ -2221,6 +2261,26 @@ Enable/Disable autostate on the SVI interface. Valid values are 'true',
 ###### `svi_management`
 Enable/Disable management on the SVI interface. Valid values are 'true', 'false', and 'default'.
 
+##### HSRP config attributes
+
+##### `hsrp_bfd`
+Enable HSRP BFD on this interface. Valid values are true, false or 'default'.
+
+##### `hsrp_delay_minimum`
+Hsrp intialization minimim delay in seconds. Valid values are integer, keyword 'default'
```
Comment (mikewiebe) [inline README.md]: `Hsrp` --> `HSRP`

File: README.md [hunk 4]
```diff
@@ -2221,6 +2261,26 @@ Enable/Disable autostate on the SVI interface. Valid values are 'true',
...
+##### HSRP config attributes
+
+##### `hsrp_bfd`
+Enable HSRP BFD on this interface. Valid values are true, false or 'default'.
+
+##### `hsrp_delay_minimum`
+Hsrp intialization minimim delay in seconds. Valid values are integer, keyword 'default'
+
+##### `hsrp_delay_reload`
+Hsrp intialization delay after reload in seconds. Valid values are integer, keyword 'default'
```
Comment (mikewiebe) [inline README.md]: `Hsrp` --> `HSRP`

File: README.md [hunk 5]
```diff
@@ -2221,6 +2261,26 @@ Enable/Disable autostate on the SVI interface. Valid values are 'true',
...
+
+##### `hsrp_mac_refresh`
+Hsrp mac refresh time in seconds. Valid values are integer, keyword 'default'
```
Comment (mikewiebe) [inline README.md]: `Hsrp` --> `HSRP`

File: README.md [hunk 6]
```diff
@@ -2221,6 +2261,26 @@ Enable/Disable autostate on the SVI interface. Valid values are 'true',
...
+
+##### `hsrp_use_bia`
+HSRP uses this interface's burned in address. Valid values are 'use_bia', 'use_bia_intf' or 'default'.
+
+##### `hsrp_version`
+Hsrp version for this interface. Valid values are integer, keyword 'default'.
```
Comment (mikewiebe) [inline README.md]: `Hsrp` --> `HSRP`

File: tests/beaker_tests/cisco_hsrp/test_interface_hsrp.rb [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  teardown { cleanup(agent) }
+  cleanup(agent)
+
+  # -------------------------------------------------------------------
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+
+  test_harness_run(tests, :default)
+
+  id = :default
+  tests[id][:desc] = '1.4 Common Defaults (absent)'
```
Comment (mikewiebe) [inline tests/beaker_tests/cisco_hsrp/test_interface_hsrp.rb]: What happened to section 1.1, 1.2 and 1.3 :)

**Comment by mikewiebe** [comment]
@saichint Looks good.  Just a few minor comments.  Also, there are some merge conflicts because I merged the code in `master` back into `develop` so those will need to be resolved before we can merge this PR.

**Comment by saichint** [comment]
Done with changes

**Comment by mikewiebe** [comment]
👍

**Pull request merged**